cmd/principal/main.go

```diff
@@ -275,6 +280,9 @@ func NewPrincipalRunCommand() *cobra.Command {
 	command.Flags().BoolVar(&enableResourceProxy, "enable-resource-proxy",
 		env.BoolWithDefault("ARGOCD_PRINCIPAL_ENABLE_RESOURCE_PROXY", true),
 		"Whether to enable the resource proxy")
+	command.Flags().DurationVar(&waitDurationForAgentPing, "wait-duration-for-agent-ping",
+		env.ParseDuration(env.StringWithDefault("ARGOCD_PRINCIPAL_WAIT_DURATION_FOR_AGENT_PING", nil, "")),
+		"Minimum time duration agent should wait before next ping to principal") // It should be less than "keep-alive-ping-interval" of agent
```

Comment:
Since this option will have dire consequences for agents (e.g. their connections will be terminated if they don't abide by this enforcement rule), we should call that out in the option's description. For example, `Drop agent connections that send keepalives more often than the specified interval`